corsikaio/subblocks/event_header.py

```diff
@@ -1,177 +1,157 @@
 import warnings
-from collections import defaultdict
+from functools import lru_cache
```

Comment:
```suggestion
from functools import cache
```

We know that we expect few versions probably only 1 per program execution, so no need to evict I think.